test/spec/app_spec.rb

```diff
@@ -585,28 +580,34 @@
 
     it "should add and remove source languages", bin6: true  do
       api_create_team_project_and_source_and_redirect_to_source('GOT', 'https://twitter.com/GameOfThrones')
-      sleep 5
-      element = @driver.find_element(:class, "source__edit-button")
+      wait_for_selector("source__tab-button-account",:class)
+      element = wait_for_selector("source__edit-button",:class)
       element.click
-      sleep 1
-      @driver.find_element(:class, "source__edit-addinfo-button").click
-      sleep 1
-      @driver.find_element(:class, "source__add-languages").click
+      wait_for_selector("source__edit-buttons-cancel-save",:class)
+      element = wait_for_selector("source__edit-addinfo-button",:class)
+      element.click
+      sleep 2
+      element = wait_for_selector("source__add-languages",:class)
+      element.click
+      sleep 2
       fill_field("sourceLanguageInput", "Acoli", :id)
       @driver.action.send_keys(:down).perform
       @driver.action.send_keys(:return).perform
-      sleep 2
+      sleep 1
+      wait_for_size_change(0, "sourceLanguageInput",:id)
       @driver.navigate.refresh
-      sleep 3
+      sleep 2
+      wait_for_selector("source-tags__tag",:class)  
       expect(@driver.page_source.include?('Acoli')).to be(true)
```

Comment:
Shouldn't we copy (not move) the line `expect(@driver.page_source.include?('Acoli')).to be(true)` to after line 597, before the `@driver.navigate.refresh`?

Reply:
The idea is to see if the tag with the language is being viewed in the page not in the edit mode.

Reply:
But after we press "Enter" we exit edit mode. I mean that we don't need to refresh the page to see that the item was actually saved. We can just press "Save". Here: https://www.youtube.com/watch?v=a7w4qDacORA

Reply:
:+1: 
Replacing the refresh by press save